src/cdk/activation.ts

```diff
@@ -4,19 +4,35 @@
  */
 
 import * as vscode from 'vscode'
+import { registerCommand } from '../shared/telemetry/telemetryUtils'
+import { AwsCdkExplorer } from './explorer/awsCdkExplorer'
 import { cdk } from './globals'
 
 /**
  * Activate AWS CDK related functionality for the extension.
  */
 export async function activate(activateArguments: { extensionContext: vscode.ExtensionContext }): Promise<void> {
-    initializeIconPaths(activateArguments.extensionContext)
+    const explorer = new AwsCdkExplorer()
+
+    await initializeIconPaths(activateArguments.extensionContext)
+
+    await registerCdkCommands(explorer)
+    activateArguments.extensionContext.subscriptions.push(
+        vscode.window.registerTreeDataProvider(explorer.viewProviderId, explorer)
+    )
 }
 
-function initializeIconPaths(context: vscode.ExtensionContext) {
+async function initializeIconPaths(context: vscode.ExtensionContext) {
```

Comment:
Is anything being awaited in this function? the `async` may be unnecessary

Reply:
fair point will remove